### PR TITLE
Include the tribe_post_count() function

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#29323](http://central.tri.be/issues/29323)
---

I needed this method to speed up some stuff on RER, so I added it in common because I've seen us use a "Modified" copy of the `wp_count_post` that WP Core has, which is not good at all.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [post.php:21](https://github.com/moderntribe/tribe-common/blob/cf1d12d7abf8e8c097397cd2c2a55dde2186d995/src/functions/post.php#L21)
:x: | Each line in an array declaration must end in a comma | [post.php:22](https://github.com/moderntribe/tribe-common/blob/cf1d12d7abf8e8c097397cd2c2a55dde2186d995/src/functions/post.php#L22)
:x: | Each line in an array declaration must end in a comma | [post.php:23](https://github.com/moderntribe/tribe-common/blob/cf1d12d7abf8e8c097397cd2c2a55dde2186d995/src/functions/post.php#L23)
:x: | String &quot; AND &quot; does not require double quotes; use single quotes instead | [post.php:103](https://github.com/moderntribe/tribe-common/blob/cf1d12d7abf8e8c097397cd2c2a55dde2186d995/src/functions/post.php#L103)

> via jenkins/codesniffer-phpcs

File: src/functions/post.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
+<?php
+if ( ! function_exists( 'tribe_post_count' ) ) {
+	/**
+	 * Counts the amount of posts we have for a given Series, it's important to note this will be cached
```
Comment (borkweb) [inline src/functions/post.php]: After reading through this function, it looks like you can genericize _Series_ down to _post_type_.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | String &quot; AND &quot; does not require double quotes; use single quotes instead | [post.php:103](https://github.com/moderntribe/tribe-common/blob/5afca5d4649809937b6e764cb0dd9eced0a38ebe/src/functions/post.php#L103)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | String &quot; AND &quot; does not require double quotes; use single quotes instead | [post.php:103](https://github.com/moderntribe/tribe-common/blob/d9a5fd76ee47800f0ed214203dd0f80f6611bf3a/src/functions/post.php#L103)

> via jenkins/codesniffer-phpcs

**Comment by bordoni** [comment]
Ready for another 👁 👁

**Pull request merged**